### DISCUSSION TITLE
Need suggestions in preparing vocab file for Training F5 TTS for Hindi language dataset

### DISCUSSION THREAD

**Discussion by abhisirka2001** [SWivid/F5-TTS#265] [Q&A]
Hey all,

I have around 1500hrs of labelled hindi language data and i want to finetune the model . I am facing issues in preparing the vocab file for Hindi. 
Any suggestions on what to change in the scrip or something which can help me in this regard?

Thanks

**Comment by dagdun** [comment]
if you want to finetune from pretrained model (recommended). 
you can modify vocab.txt by replacing it with your language into vocab.txt.
be careful to maintain the total number of characters at 2,545, same as the original. [#221](https://github.com/SWivid/F5-TTS/issues/221#issuecomment-2431008184)
if you train the model from scratch you can create your own vocab.txt.

the batch_size_per_gpu setting, compare it with the configuration of the A100x8.

For me, the test at step 58,000 can speak my language (but very poorly).

Tip: After you finish creating your vocab.txt, replace the existing vocab.txt in all relevant parts of the project.

**Reply by abhisirka2001** [reply]
Thanks 
Really Helpful!

**Comment by NehaB18** [comment]
@dagdun did you manage to train the a good model? how much time and data did it take?

**Reply by dagdun** [reply]
I believe my model can speak 93% of our language. However, a limitation of this solution is that it calculates speaking durations based on the number of characters. sometimes, it skips words because there are too many to speak but fewer characters.
for the best result in my language, the reference voice should be 4-6 seconds long. the output voice should be 3-8 seconds and 0.9 speed. Use multiple styles if I want to generate longer text.

Other information about training:
- 150,000 clips ranging from 2 to 15 seconds from Mozilla Common Voice (Thai language only)
- Using Google Cloud GPU 1xL4 (total cost around $160)
- Batch size: 82,000 (increased to use almost 90% of VRAM)
- Total steps: 200,000
- Training took 8 days, including setup and experimentation.

**Reply by NehaB18** [reply]
Thanks, very helpful.

**Reply by asfandsheraz** [reply]
hi how can we have a hindi tranned tts?

**Comment by x4080** [comment]
What if I wanted to train only the language (using alphabet) like for malaysian, indonesian etc ? What data should I prepare ?
